Add docker release docs (#1674)
elastic/connectors-python: docs/RELEASING.md

- Update the **BUILD** version ([example PR](https://github.com/elastic/connectors-python/pull/122)). Note that the Connectors project does not immediately bump to the next **PATCH** version. That won't happen until that patch release's FF date.

## Releasing docker images

To release the docker image, follow these steps:

1. Make sure you're on the right tagged release commit, e.g. `v8.10.2.0`

>    - If no release tag exists yet, use the release branch, e.g. `8.10`

Reviewer: ```suggestion
   - If no release tag exists yet, create it with `git tag <tagid> && git push origin <tagid>`
```

2. Make sure the version in [VERSION](../connectors/VERSION) is correct

> 3. Run `docker login -u <username> -p <password> docker.elastic.co` with credentials that allow for release

Reviewer 2: Is it just `allow releasing` instead of `allow for release`?

Asking as a non-native, who barely sees "allow for" 😅

Author: As a native speaker with a weird "it just works to me" understanding; I took a look to see why `allow for` felt right to me and found this post: https://english.stackexchange.com/questions/30069/whats-the-difference-between-allow-and-allow-for

So in this sentence, `that allow for releasing` is referring to `credentials` rather than the run command. Having the credentials does not automatically release anything, so `allows for` works.
The difference is very small, but to me `allow releasing` would work if it was referencing the command rather than the credentials, as running the command directly results in a release.

Reviewer 2: Thanks for the explanation!

> 4. Edit the [Makefile](../Makefile) to remove `-SNAPSHOT` from the `docker-build`, `docker-run` and `docker-push` steps

Reviewer 3: Should we add a `docker-release` target?

Author: @wangch079 we aren't expecting to need to run this manually because it's automated, so I thought some instructions would be enough rather than a new makefile target.

Reviewer 2: Is there a ticket for automation already?

Author: @sphilipse do you know if automating this in 8.11 has a ticket?
